src/mongocxx/database.cpp

```diff
@@ -97,6 +97,11 @@ class collection database::create_collection(stdx::string_view name,
     return mongocxx::collection(*this, static_cast<void*>(result));
 }
 
+void database::drop() {
+    bson_error_t error;
+    libmongoc::database_drop(_impl->database_t, &error);
```

Comment:
you are going to want to test the error after this call, and then throw an exception if it is non-OK.